oneargopy/Argo.py

```diff
@@ -227,54 +345,432 @@ def __load_sprof_dataframe(self) -> pd:
+        
+        # Filter by other constraints, these functions will use self.selection_frame 
+        # so we don't have to pass a frame
+        if self.ocean : self.__get_in_ocean_basin()
+        # other narrowing functions that act on created selection frame...
+
+        # Convert the working dataframe into a dictionary
+        selected_floats_dict = self.__dataframe_to_dictionary()
+
+        # Printing Dict, likely to remove after testing period
+        print(f'Floats: {selected_floats_dict.keys()}')
+        for key, value in selected_floats_dict.items():
+            print(f'{key}: {value}')
+
+        return selected_floats_dict
+        
+
+    def __get_in_geographic_range(self, dataframe_to_filter: pd)-> list:
+        """ A function to create and return two true false arrays indicating
+            floats and profiles that fall within the geographic range.
+        """
+        # If the user has passed us the entire globe don't go through the whole
+        # process of checking if the points of all the floats are inside the polygon
+        if self.keep_full_geographic: 
+            return  [True] * len(dataframe_to_filter)
+        
+        if self.download_settings.verbose: print(f'Sorting floats for those within the geographic range...')
+
+        # Make points out of profile lat and lons
+        if self.download_settings.verbose: print(f'Creating point list from profiles...')
+        profile_points = np.empty((len(dataframe_to_filter), 2))
+        
+        # The longitudes in the dataframe are standardized to fall within -180 and 180.
+        # but our longitudes only have a standard minimum value of -180. In this section
+        # we adjust the longitude and latitudes in the dataframe to follow this minimum 
+        # only approach.
+        if max(self.lon_lim) > 180:
+            if self.download_settings.verbose: print(f'The max value in lon_lim is {max(self.lon_lim)}')
+            if self.download_settings.verbose: print(f'Adjusting longitude values...')
+            profile_points[:,0] = dataframe_to_filter['longitude'].apply(lambda x: x + 360 if -180 < x < min(self.lon_lim) else x).values
+        else:
+            profile_points[:,0] = dataframe_to_filter['longitude'].values
+        
+        # Latitudes in the dataframe are good to go
+        profile_points[:,1] = dataframe_to_filter['latitude'].values
+
+        # Create polygon or box using lat_lim and lon_lim 
+        if self.download_settings.verbose: print(f'Creating polygon...')
+        if len(self.lat_lim) == 2:
+            shape = [[max(self.lon_lim), min(self.lat_lim)], # Top-right
+                     [max(self.lon_lim), max(self.lat_lim)], # Bottom-right
+                     [min(self.lon_lim), max(self.lat_lim)], # Bottom-left
+                     [min(self.lon_lim), min(self.lat_lim)]] # Top-left
+        else:
+            shape = []
+            for lat, lon in zip(self.lat_lim, self.lon_lim):
```

Comment:
reverse lat,lon both times to match "shape"

Reply:
![image](https://github.com/lovelysandlonelys/OneArgo-Python-Dev/assets/91633941/af9271f9-3007-45bf-bcd7-724d58f5e1cf)